package.json

```diff
@@ -84,14 +84,11 @@
     "stream-to-string": "^1.2.1",
     "streamify-array": "^1.0.1",
     "streamify-string": "^1.0.1",
-    "ts-jest": "^29.1.1",
+    "ts-jest": "^29.0.0",
     "ts-loader": "^9.4.4",
     "typedoc": "^0.26.0",
-    "typescript": "^5.3.3",
+    "typescript": "~5.5.0",
```

Comment:
Why the `~`?

Reply:
The newer TS compiler versions seem to dislike something in the `lru-cache` package, if I remember correctly, that is pulled in as a dependency of a dependency. The build would fail with typing errors for that package.

The reason the master branch works at the moment is because the TS version is locked in the Yarn lockfile. Bumping the TS version will introduce those errors, and it has been like this for a couple of months now (since version 5.6 was released, iirc, since the TS version is locked to 5.something).

Reply:
I remember @smessie running into this recently.
What was your solution again @smessie?

Reply:
In any case, we usually use yarn resolutions for locking to a specific version in the monorepo. So if needed, let's use that.

Reply:
Yes, it was because of an issue with lru-cache and TS, explained here https://github.com/isaacs/node-lru-cache/issues/354#issuecomment-2351174972
I don't remember the exact location I encountered it, but we concluded that the issue was fixed in the newer lru-cache version, but that specific version does no longer support Node 18. With Comunica, we didn't want to drop support for Node 18 yet, as it only reaches its end of life by the end of April 2025. The idea was to then drop Node 18 support and update the lru-cache dependency.
A temporary workaround/fix was to downgrade TypeScript to 5.5.4.

Reply:
Right, thanks for the reminder @smessie! (and good that we have this written down here for future reference)